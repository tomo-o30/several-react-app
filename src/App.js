import { useState } from "react";
import "./App.css";

function App() {
  const [todos, setTodos] = useState([]);
  const [todo, setTodo] = useState("");
  const [completeFlag, setCompleteFlag] = useState(false);

  const handleInputChange = (e) => {
    setTodo(e.target.value);
  };

  const handleButtonClick = (e) => {
    e.preventDefault();

    if (todo !== "") {
      setTodos([
        ...todos,
        {
          id: todos.length + 1,
          completeFlag: completeFlag,
          text: todo.trim(),
        },
      ]);
      setTodo("");
    }
  };

  return (
    <div>
      <div className="input-area">
        <input
          name="todo"
          type="text"
          placeholder="Create a new todo"
          value={todo}
          onChange={handleInputChange}
        />
        <button onClick={handleButtonClick}>追加</button>
      </div>
      <div className="imcomplete-area">
        <p className="title">未完了のTODO</p>
        <ul>
          {todos
            .filter((todo) => todo.completeFlag === false)
            .map((todo) => (
              <>
                <li key={todo.id}>{todo.text}</li>
                <button>追加</button>
                <button>削除</button>
              </>
            ))}
        </ul>
      </div>
      <div className="complete-area">
        <p className="title">完了のTODO</p>
        <ul>
          {todos
            .filter((todo) => todo.completeFlag === true)
            .map((todo) => (
              <>
                <li key={todo.id}>{todo.text}</li>
                <button>追加</button>
                <button>削除</button>
              </>
            ))}
        </ul>
      </div>
    </div>
  );
}

export default App;
