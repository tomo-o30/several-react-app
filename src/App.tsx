import { useState } from "react";
import "./App.css";
import Button from "./components/Button";

interface Todo {
  id: number;
  completeFlag: boolean;
  text: string;
}

const App = ()=> {
  const [todos, setTodos] = useState<Todo[]>([]);
  const [todo, setTodo] = useState("");
  const [id, setId] = useState(0);

  const handleInputChange = (e:React.ChangeEvent<HTMLInputElement>) => {
    setTodo(e.target.value);
  };

  const handleButtonClick = (e?:React.MouseEvent<HTMLButtonElement>) => {
    if (e) {
      e.preventDefault();
    }

    if (todo !== "") {
      setId((prevId) => prevId + 1);
      setTodos([
        ...todos,
        {
          id,
          completeFlag: false,
          text: todo.trim(),
        },
      ]);
      console.log(todos);
      setTodo("");
    }
  };

  const handleComplete = (id:number) => {
    const updateTodos = todos.map((todo) => {
      if (todo.id === id) {
        return { ...todo, completeFlag: true };
      } else {
        return todo;
      }
    });
    setTodos(updateTodos);
  };

  const handleDelete = (id:number) => {
    const newTodos = todos.filter((todo) => {
      return todo.id !== id;
    });
    setTodos(newTodos);
  };

  const handleBeBack = (id:number) => {
    const updateTodos = todos.map((todo) => {
      if (todo.id === id) {
        return { ...todo, completeFlag: false };
      } else {
        return todo;
      }
    });
    setTodos(updateTodos);
  };

  return (
    <div>
      <div className="input-area">
        <input
          name="todo"
          type="text"
          placeholder="Create a new todo"
          value={todo}
          onChange={handleInputChange}
        />
        <Button onClick={(e)=>handleButtonClick(e)} text='追加' />
      </div>
      <div className="imcomplete-area">
        <p className="title">未完了のTODO</p>
        <ul>
          {todos
            .filter((todo) => todo.completeFlag === false)
            .map((todo) => (
              <li className="todo-area" key={todo.id}>
                {todo.text}
                <Button onClick={() => handleComplete(todo.id)} text='完了'/>
                <Button onClick={() => handleDelete(todo.id)} text='削除'/>
              </li>
            ))}
        </ul>
      </div>
      <div className="complete-area">
        <p className="title">完了のTODO</p>
        <ul>
          {todos
            .filter((todo) => todo.completeFlag === true)
            .map((todo) => (
              <li className="todo-area" key={todo.id}>
                {todo.text}{" "}
                <Button onClick={() => handleBeBack(todo.id)} text='戻す'/>
              </li>
            ))}
        </ul>
      </div>
    </div>
  );
}

export default App;
