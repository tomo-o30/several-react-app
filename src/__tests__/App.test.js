import renderer from "react-test-renderer";
import App from "../pages/App";

it("changes the class when hovered", () => {
  const component = renderer.create(<App />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
