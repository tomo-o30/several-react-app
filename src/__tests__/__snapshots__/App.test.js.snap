// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes the class when hovered 1`] = `
<div>
  <div
    className="input-area"
  >
    <input
      name="todo"
      onChange={[Function]}
      placeholder="Create a new todo"
      type="text"
      value=""
    />
    <button
      onClick={[Function]}
    >
      追加
    </button>
  </div>
  <div
    className="imcomplete-area"
  >
    <p
      className="title"
    >
      未完了のTODO
    </p>
    <ul />
  </div>
  <div
    className="complete-area"
  >
    <p
      className="title"
    >
      完了のTODO
    </p>
    <ul />
  </div>
</div>
`;
